src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -944,7 +907,7 @@ void c_typecheck_baset::typecheck_expr_sizeof(exprt &expr)
   }
   else if(expr.operands().size()==1)
   {
-    type.swap(expr.op0().type());
+    type.swap(to_unary_expr(expr).op().type());
```

Comment:
Is the `else` case actually possible? (Parser-provided guarantees and such.) If that isn't the case, then the `else if` should become the `else`.

Reply:
Indeed, the else should not happen.